gateway-sp-comms/src/single_sp.rs

```diff
@@ -723,6 +743,12 @@ impl Inner {
     }
 
     async fn run(mut self) {
+        // If discovery fails (typically due to timeout, but also possible due
+        // to misconfiguration where we can't send packets at all), how long do
+        // we wait before retrying? If failure is due to misconfiguration, we
+        // will never succeed - how do we cope with that?
```

Comment:
I think this latter part is something we'll have to deal with via metrics/alerts at some point.